### PR TITLE
[APP] Frontend config improvement

### PR DESCRIPTION
**Pull request by udai-arneja**
# Summary

Improved Advanced Settings and normal Settings. Scrollable and fits the screen better (columns etc). More modular.
Opinons on removing advanced settings or settings altogether as we probably won't have the health (tau etc) stuff so no real real need for having a difference.

sorry about messing around with the API file, I reverted that change pls double check

### CODE REVIEW

File: frontend/src/Components/NewRunConfig/AdvancedSettings/AdvancedSettings.tsx [hunk 1]
```diff
@@ -13,88 +16,33 @@ interface AdvancedSettingsProps {
 export default function AdvancedSettings(props: AdvancedSettingsProps) {
   const { config, setConfig } = props;
 
-  const [disableTotalFood, setDisableTotalFood] = useState(Boolean);
-
   function configHandler<Key extends keyof SimConfig>(value: any, keyString: any) {
     var key: Key = keyString; // converting keyString to type Key
     config[key] = value;
     setConfig(config);
+    console.log(config);
```
Comment (JaafarRammal) [inline frontend/src/Components/NewRunConfig/AdvancedSettings/AdvancedSettings.tsx:23]: Could keep could remove? Could be useful when things crash

File: frontend/src/Components/NewRunConfig/ParameterGroups/ParameterLabels.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+export interface Parameter {
+  helperText: string;
+  label: string;
+  labelFor: string;
+  labelInfo: string;
+  key: keyof(SimConfig);
+  min: number;
+}
+
+// might not need these
```
Comment (JaafarRammal) [inline frontend/src/Components/NewRunConfig/ParameterGroups/ParameterLabels.ts:12]: Are we able to remove then?

File: frontend/src/Components/NewRunConfig/ParameterGroups/TowerLength.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    labelFor: "text-input",
+    labelInfo: "",
+    key: "ReshuffleDays",
+    min: 1,
+  },
+];
+
+export const lengthParamsAdv: Parameter[] = [
+  {
+    helperText: "Agents can do one/two actions per 'Tick'",
```
Comment (jzzheng22) [inline frontend/src/Components/NewRunConfig/ParameterGroups/TowerLength.tsx:68]: Don't like this. Prob just get rid of it if possible.

**Review by jzzheng22 — CHANGES_REQUESTED** [review]
The 'Advanced Run' view is a lot nicer than the 'Quick Run' view imo - would prefer if you get the two to look the same. Also I don't see the point in having the two views - they look almost identical?

File: frontend/src/Components/NewRunConfig/ParameterGroups/TowerLength.tsx (hunk 3 above)
Comment (jzzheng22) [reply frontend/src/Components/NewRunConfig/ParameterGroups/TowerLength.tsx:68]: (for clarity; i mean the helperText description. It's not entirely true.

File: frontend/src/Components/NewRunConfig/ParameterGroups/LogDescription.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+              }}
+            />
+          </FormGroup>
+        </div>
+        <div className="col-lg-6 d-flex justify-content-center">
+          <FormGroup label="File Name" labelFor="text-input" key="FileName">
+            <input
+              type="text"
+              onChange={(value) => configHandler(value.target.value, "LogFileName")}
+              placeholder="Simulation Run #"
```
Comment (JaafarRammal) [inline frontend/src/Components/NewRunConfig/ParameterGroups/LogDescription.tsx:25]: Can we change this to filename actually? :)

**Review by JaafarRammal — APPROVED** [review]
LGTM, especially breaking it into different small files. A few minor comments that you can resolve directly if no changes required

One note for conditional classes: you can use this form inside the div directly:
```ts
var classNameString = `${advanced ? "col-lg-6" : "col-lg-4"} d-flex justify-content-center`;
```

File: frontend/src/Components/NewRunConfig/ParameterGroups/TowerLength.tsx (hunk 3 above)
Comment (udai-arneja) [reply frontend/src/Components/NewRunConfig/ParameterGroups/TowerLength.tsx:68]: How do we indicate what unit of measurement a tick is ? @jzzheng22

Comment (udai-arneja) [reply frontend/src/Components/NewRunConfig/ParameterGroups/TowerLength.tsx:68]: I wasn't too sure but wanted to provide some context...

**Comment by udai-arneja** [comment]
@jzzheng22 im thinking of removing new run completely, is that cool?

File: frontend/src/Components/NewRunConfig/AdvancedSettings/AdvancedSettings.tsx (hunk 1 above)
Comment (udai-arneja) [reply frontend/src/Components/NewRunConfig/AdvancedSettings/AdvancedSettings.tsx:23]: Will remove.

File: frontend/src/Components/NewRunConfig/ParameterGroups/ParameterLabels.ts (hunk 2 above)
Comment (udai-arneja) [reply frontend/src/Components/NewRunConfig/ParameterGroups/ParameterLabels.ts:12]: Yep.

**Comment by jzzheng22** [comment]
> @jzzheng22 im thinking of removing new run completely, is that cool?

What do you mean by 'new run'?

**Comment by udai-arneja** [comment]
The quick run config on the side. Completely remove it. Just have the advanced.

**Comment by jzzheng22** [comment]
> The quick run config on the side. Completely remove it. Just have the advanced.

Yup I'd agree, they're mostly the same anyway.

File: frontend/src/Components/NewRunConfig/ParameterGroups/TowerLength.tsx (hunk 3 above)
Comment (jzzheng22) [reply frontend/src/Components/NewRunConfig/ParameterGroups/TowerLength.tsx:44]: Can we just leave it blank?

**Review by jzzheng22 — APPROVED** [review]
Looks good to me.

**Comment by jzzheng22** [comment]
Also ideally we change some of the default values to be more useful, but we can do that in a separate PR.

File: frontend/src/Components/Navbar.tsx [hunk 5]
```diff
@@ -12,17 +12,9 @@ export default function Nav() {
         <Button
           className="bp3-minimal"
           icon="search-around"
-          text="Quick Run"
+          text="New Run Submission"
```
Comment (JaafarRammal) [inline frontend/src/Components/Navbar.tsx:15]: Just "New Run" imo

**Review by JaafarRammal — APPROVED** [review]
LGTM!!
Feel free to resolve comment

**Pull request merged**